src/helper/subscriber/waf.cpp

```diff
@@ -188,10 +249,11 @@ dds::result instance::listener::call(dds::parameter &data, unsigned timeout) {
         throw invalid_object();
     case DDWAF_ERR_INVALID_ARGUMENT:
         throw invalid_argument();
-    case DDWAF_ERR_TIMEOUT:
```

Comment:
I don't think it's a good idea to remove this case. Even if now we never pass 0 as budget, subsequent changes in the WAF or the helper may cause that to happen.

Reply:
If you recall we removed `DDWAF_ERR_TIMEOUT` in favour of the timeout flag in `ddwaf_result` in order to track all possible timeouts, we're now throwing the exception in the `DDWAF_GOOD` case when `result.timeout` is set.

Reply:
I don't recall, but it all makes sense now.